Source/DafnyLanguageServer/Workspace/LanguageServerExtensions.cs

```diff
@@ -32,7 +32,8 @@ public static class LanguageServerExtensions {
         .AddSingleton<ITextChangeProcessor, TextChangeProcessor>()
         .AddSingleton<ISymbolTableRelocator, SymbolTableRelocator>()
         .AddSingleton<ISymbolGuesser, SymbolGuesser>()
-        .AddSingleton<ICompilationStatusNotificationPublisher, CompilationStatusNotificationPublisher>();
+        .AddSingleton<ICompilationStatusNotificationPublisher, CompilationStatusNotificationPublisher>()
+        .AddSingleton<IDiagnosticPublisher, DiagnosticPublisher>();
```

Comment:
Did you really mean to add a second `IDiagnosticPublisher` here?  There's already one above.

Reply:
Nope, can be removed safely. Actually I'm surprised that .NET Core didn't complain about this. But might be because I'm registering the same type.